huaweicloud/services/workspace/resource_huaweicloud_workspace_app_nas_storage.go

```diff
@@ -0,0 +1,290 @@
+	}
+
+	createPath := client.Endpoint + httpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", client.ProjectID)
+	createOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         utils.RemoveNil(buildAppNasStorageCreateOpts(d)),
+	}
+	requestResp, err := client.Request("POST", createPath, &createOpt)
+	if err != nil {
+		return diag.Errorf("error creating NAS storage of Workspace APP: %s", err)
+	}
+	respBody, err := utils.FlattenResponse(requestResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	storageId := utils.PathSearch("id", respBody, "").(string)
+	if storageId == "" {
+		return diag.Errorf("unable to find the storage ID from the API response")
+	}
+	d.SetId(storageId)
+
+	return resourceAppNasStorageRead(ctx, d, meta)
+}
+
+func flattenStorageMetadata(metadata interface{}) []map[string]interface{} {
+	if metadata != nil {
+		return []map[string]interface{}{
+			{
+				"storage_handle":  utils.PathSearch("storage_handle", metadata, nil),
+				"storage_class":   utils.PathSearch("storage_class", metadata, nil),
+				"export_location": utils.PathSearch("export_location", metadata, nil),
+			},
+		}
+	}
+	return nil
+}
+
+func GetAppNasStorageById(client *golangsdk.ServiceClient, storageId string) (interface{}, error) {
+	httpUrl := "v1/{project_id}/persistent-storages?storage_id={storage_id}"
+
+	getPath := client.Endpoint + httpUrl
+	getPath = strings.ReplaceAll(getPath, "{project_id}", client.ProjectID)
+	getPath = strings.ReplaceAll(getPath, "{storage_id}", storageId)
+	getOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	requestResp, err := client.Request("GET", getPath, &getOpt)
+	if err != nil {
+		return nil, err
+	}
+	respBody, err := utils.FlattenResponse(requestResp)
+	if err != nil {
+		return nil, err
+	}
+	storageConfig := utils.PathSearch("items|[0]", respBody, nil)
+	if storageConfig == nil {
+		return nil, golangsdk.ErrDefault404{}
+	}
+	return storageConfig, nil
+}
+
+func resourceAppNasStorageRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg       = meta.(*config.Config)
+		region    = cfg.GetRegion(d)
+		storageId = d.Id()
+	)
+
+	client, err := cfg.NewServiceClient("appstream", region)
+	if err != nil {
+		return diag.Errorf("error creating Workspace APP client: %s", err)
+	}
+
+	storageConfig, err := GetAppNasStorageById(client, storageId)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "NAS Storage of Workspace APP")
+	}
+
+	mErr := multierror.Append(nil,
+		d.Set("region", region),
+		d.Set("name", utils.PathSearch("name", storageConfig, nil)),
+		d.Set("storage_metadata", flattenStorageMetadata(utils.PathSearch("storage_metadata", storageConfig, nil))),
+		d.Set("created_at", utils.FormatTimeStampRFC3339(utils.ConvertTimeStrToNanoTimestamp(utils.PathSearch("create_time",
+			storageConfig, "").(string))/1000, false)),
+	)
+	if err = mErr.ErrorOrNil(); err != nil {
```

Comment:
```suggestion
	return diag.FromErr(mErr.ErrorOrNil())
```

Reply:
seems the error message is missing, fixed.